backend/azure-functions/Services/ExampleDataService.ts

```diff
@@ -0,0 +1,33 @@
+import { IExampleDataService } from "./IExampleDataService";
+import { IExample } from "../Models/IExample";
+import { ICollection } from "./ICollection";
+import { InsertFailedError } from "../Models/InsertFailedError";
+
+export class ExampleDataService implements IExampleDataService {
```

Comment:
```suggestion
export class ExampleDataService implements IExampleDataService {
```
i'd to avoid 2 different abstractions of the database (the second is mongo collection creator),
lets try to have 1 provider that read/write any data